src/map_locations.c

```diff
@@ -93,14 +94,25 @@ TbBool get_coords_at_meta_action(struct Coord3d *pos, PlayerNumber target_plyr_i
     case MML_RECENT_COMBAT:
         src = &dungeon->last_combat_location;
         break;
+    case MML_ACTIVE_CTA:
+        targetpos = malloc(sizeof(struct Coord3d));
+        if (targetpos == NULL) {
+            // Handle memory allocation failure
+            return false;
+        }
+        targetpos->x.val = subtile_coord_center(dungeon->cta_stl_x);
+        targetpos->y.val = subtile_coord_center(dungeon->cta_stl_y);
+        targetpos->z.val = get_floor_height_at(pos);
+        src = targetpos; 
```

Comment:
what's with the targetpos malloc instead of simply just assigning to src directly?

Reply:
oh right because triggered_object_location and last_combat_location are used as is, but a pointer to a stack variable still makes more sense then doing it on the heap